sa/storage-authority.go

```diff
@@ -237,6 +237,46 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64
 	return ssa.GetAuthorization(auth.ID)
 }
 
+type TooManyCertificatesError string
+
+func (t TooManyCertificatesError) Error() string {
+	return string(t)
+}
+
+// CountCertificates returns the number of certificates issued within a time
+// period containing DNSNames that are equal to, or subdomains of, the given name.
+func (ssa *SQLStorageAuthority) CountCertificatesByName(domain string, earliest, latest time.Time) (int, error) {
+	var count int64
+	const max = 10000
```

Comment:
This needs to be documented, or else we could write a nice for-loop to handle the arbitrary amount.

I'm a little worried about not being able to count higher than 10000 but I guess we'll have to rely on the rate limit being orders of magnitude lower than that.